### PR TITLE
CLOUDP-174790 Add command "atlas dataLakePipelines datasets delete"

### PR DESCRIPTION
**Pull request by fmenezes**
## Proposed changes

Add command "atlas dataLakePipelines datasets delete"

_Jira ticket:_ CLOUDP-174790

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

**Review by swattyT — APPROVED** [review]
LGTM!

File: internal/store/data_lake_pipelines_datasets.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+// DeletePipelineDataset encapsulates the logic to manage different cloud providers.
+func (s *Store) DeletePipelineDataset(projectID, pipelineName, id string) error {
+	switch s.service {
+	case config.CloudService, config.CloudGovService:
+		_, _, err := s.clientv2.DataLakePipelinesApi.DeletePipelineRunDataset(s.ctx, projectID, pipelineName, id).Execute()
+		return err
+	default:
+		return fmt.Errorf("%w: %s", errUnsupportedService, s.service)
+	}
```
Comment (matt-condon) [inline internal/store/data_lake_pipelines_datasets.go:39]: nit: I don't think we need the switch case for new stores that are only supported by atlas

Comment (fmenezes) [reply internal/store/data_lake_pipelines_datasets.go]: @matt-condon I've moved the stores around

**Review by matt-condon — APPROVED** [review]
LGTM 🏅

**Pull request merged**